backend/drm/drm.c

```diff
@@ -884,15 +908,24 @@ static void realloc_crtcs(struct wlr_drm_backend *drm, bool *changed_outputs) {
 
 	wlr_log(WLR_DEBUG, "Reallocating CRTCs");
 
-	uint32_t crtc[drm->num_crtcs + 1];
+	uint32_t *crtc = malloc(sizeof(uint32_t) * drm->num_crtcs);
+	if (crtc == NULL) {
+		goto error_crtc;
+	}
 	for (size_t i = 0; i < drm->num_crtcs; ++i) {
 		crtc[i] = UNMATCHED;
 	}
 
-	struct wlr_drm_connector *connectors[num_outputs + 1];
+	struct wlr_drm_connector **connectors =
+		calloc(num_outputs, sizeof(struct wlr_drm_connector));
```

Comment:
This is the wrong sizeof. This is an array of pointers, not just a normal array.